lib/block_view.go

```diff
@@ -7009,6 +8537,43 @@ func (bav *UtxoView) GetUnspentUtxoEntrysForPublicKey(pkBytes []byte) ([]*UtxoEn
 	return utxoEntriesToReturn, nil
 }
 
+func (bav *UtxoView) GetSpendableBitcloutBalanceNanosForPublicKey(pkBytes []byte,
+) (_spendableBalance uint64, _err error) {
+	// In order to get the spendable balance, we need to account for any immature block rewards.
+	// We get these by starting at the chain tip and iterating backwards until we have collected
+	// all of the immature block rewards for this public key.
+	nextBlockHash := bav.TipHash
+	numImmatureBlocks := uint64(bav.Params.BlockRewardMaturity / bav.Params.TimeBetweenBlocks)
+	immatureBlockRewards := uint64(0)
+	for ii := uint64(1); ii < numImmatureBlocks; ii++ {
+		// Don't look up the genesis block since it isn't in the DB.
+		if GenesisBlockHashHex == nextBlockHash.String() {
+			break
+		}
+
+		block, err := GetBlock(nextBlockHash, bav.Handle)
+		if err != nil || block == nil {
+			return uint64(0), errors.Wrapf(
+				err, "GetSpendableBitcloutBalanceNanosForPublicKey: Problem getting block for blockhash %s",
+				nextBlockHash.String())
+		}
+		blockRewardForPK, err := DbGetBlockRewardForPublicKeyBlockHash(bav.Handle, pkBytes, nextBlockHash)
+		if err != nil {
+			return uint64(0), errors.Wrapf(
+				err, "GetSpendableBitcloutBalanceNanosForPublicKey: Problem getting block reward for "+
+					"public key %s blockhash %s", PkToString(pkBytes, bav.Params), nextBlockHash.String())
+		}
+		immatureBlockRewards += blockRewardForPK
+		nextBlockHash = block.Header.PrevBlockHash
+	}
+
+	balanceNanos, err := bav.GetBitcloutBalanceNanosForPublicKey(pkBytes)
+	if err != nil {
+		return uint64(0), errors.Wrap(err, "GetSpendableUtxosForPublicKey: ")
+	}
+	return balanceNanos - immatureBlockRewards, nil
```

Comment:
I would sanity-check that balanceNanos > immatureBlockRewards so you don't get an underflow.

Reply:
I checked >= instead of > since they could feasibly be the same and the spendable balance would be zero.